pytorch_lightning/plugins/training_type/training_type_plugin.py

```diff
@@ -30,6 +30,8 @@
 if TYPE_CHECKING:
     from pytorch_lightning.trainer.trainer import Trainer
 
+TBroadcast = TypeVar("T", bound=object)
```

Comment:
Could this be simplified?
```suggestion
TBroadcast = TypeVar("T")
```